### PR TITLE
Update SCC.mdx

### PR DESCRIPTION
**Pull request by leoleo28**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [v] I have tested my code.
- [v] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [v] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [ ] I have linked this PR to any issues that it closes.

### CODE REVIEW

File: content/6_Advanced/SCC.mdx [hunk 1]
```diff
@@ -99,6 +99,138 @@ int main() {
...
+		List<Integer> ad = (pass == 1) ? adj[x] : adj_t[x];
+		for (int e : ad) {
+			if (!vis[e]) { dfs(e, pass, num); }
+		}
+		S.add(x);
+		if (pass == 2) id[x] = num;
+	}
+
+	// BeginCodeSnip{FastIO}
+	private static class FastIO extends PrintWriter {
```
Comment (SansPapyrus683) [inline content/6_Advanced/SCC.mdx:167]: can you use [kattio](https://usaco.guide/general/adding-solution?lang=java#general)

Comment (leoleo28) [reply content/6_Advanced/SCC.mdx]: Sure.

File: content/6_Advanced/SCC.mdx [hunk 2]
```diff
@@ -163,6 +163,33 @@ public class Main {
 		if (pass == 2) id[x] = num;
 	}
 	// CodeSnip{Kattio}
+	static class Kattio extends PrintWriter {
```
Comment (SansPapyrus683) [inline content/6_Advanced/SCC.mdx]: uh just the codesnip is fine

Comment (leoleo28) [reply content/6_Advanced/SCC.mdx]: Ok, I fix it.

File: content/6_Advanced/SCC.mdx [hunk 3]
```diff
@@ -99,6 +99,76 @@ int main() {
...
+
+```java
+
+import java.io.*;
+import java.util.*;
+
+public class Main {
+	static boolean[] vis;
+	static List<Integer>[] adj;    // // Adjacency list of neighbor
+	static List<Integer>[] adj_t;  // adj_t is the transpose of adj
```
Comment (SansPapyrus683) [inline content/6_Advanced/SCC.mdx:113]: java uses camelCase

File: content/6_Advanced/SCC.mdx [hunk 4]
```diff
@@ -99,6 +99,76 @@ int main() {
...
+	static List<Integer> S;
+	public static void main(String[] args) {
+		Kattio io = new Kattio();
+		int n = io.nextInt();
+		int m = io.nextInt();
+		adj = new ArrayList[n + 1];
+		adj_t = new ArrayList[n + 1];
+		vis = new boolean[n + 1];
+		id = new int[n + 1];
+		S = new ArrayList<>();
```
Comment (SansPapyrus683) [inline content/6_Advanced/SCC.mdx:124]: initialize variables above where you actually use them

File: content/6_Advanced/SCC.mdx [hunk 5]
```diff
@@ -99,6 +99,76 @@ int main() {
...
+				components++;
+				dfs(S.get(i), 2, components);
+			}
+		}
+
+		io.println(components);
+		for (int i = 1; i <= n; i++) io.print(id[i] + " ");
+		io.close();
+	}
+	private static void dfs(int x, int pass, int num) {
```
Comment (SansPapyrus683) [inline content/6_Advanced/SCC.mdx:156]: space out functions

File: content/6_Advanced/SCC.mdx [hunk 6]
```diff
@@ -99,6 +99,76 @@ int main() {
...
+		int components = 0;
+		for (int i = n - 1; i >= 0; i--) {
+			if (!vis[S.get(i)]) {
+				components++;
+				dfs(S.get(i), 2, components);
+			}
+		}
+
+		io.println(components);
+		for (int i = 1; i <= n; i++) io.print(id[i] + " ");
```
Comment (SansPapyrus683) [inline content/6_Advanced/SCC.mdx:153]: braces

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: content/6_Advanced/SCC.mdx (hunk 3 above)
Comment (leoleo28) [reply content/6_Advanced/SCC.mdx]: FIx it

File: content/6_Advanced/SCC.mdx (hunk 4 above)
Comment (leoleo28) [reply content/6_Advanced/SCC.mdx]: Fix it

File: content/6_Advanced/SCC.mdx (hunk 5 above)
Comment (leoleo28) [reply content/6_Advanced/SCC.mdx:155]: Fix it

File: content/6_Advanced/SCC.mdx (hunk 6 above)
Comment (leoleo28) [reply content/6_Advanced/SCC.mdx]: Fix it

File: content/6_Advanced/SCC.mdx [hunk 7]
```diff
@@ -98,6 +97,71 @@ int main() {
...
+
+public class Main {
+	static final int N = 100010;
+	static boolean[] vis = new boolean[N + 1];
+	// Adjacency list of neighbor
+	static List<Integer>[] adj = new ArrayList[N + 1];
+	 // adjT is the transpose of adj
+	static List<Integer>[] adjT = new ArrayList[N + 1];
+	static int[] id = new int[N + 1];
+	static List<Integer> S = new ArrayList<>();
```
Comment (SansPapyrus683) [inline content/6_Advanced/SCC.mdx:114]: better variable name

File: content/6_Advanced/SCC.mdx [hunk 8]
```diff
@@ -98,6 +97,71 @@ int main() {
...
 
 </CPPSection>
+<JavaSection>
+
+```java
+import java.io.*;
+import java.util.*;
+
+public class Main {
+	static final int N = 100010;
```
Comment (SansPapyrus683) [inline content/6_Advanced/SCC.mdx:107]: why is this 10 greater than the actual N

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: content/6_Advanced/SCC.mdx (hunk 7 above)
Comment (leoleo28) [reply content/6_Advanced/SCC.mdx]: Fix it

File: content/6_Advanced/SCC.mdx (hunk 8 above)
Comment (leoleo28) [reply content/6_Advanced/SCC.mdx]: Fix it

**Comment by SansPapyrus683** [comment]
can i have a screenshot of AC with the _updated_ code

**Comment by leoleo28** [comment]
> can i have a screenshot of AC with the _updated_ code included in said screenshot

Sure, the attached file is the screenshot.

截圖 2023-12-31 晚上8 08 34

**Comment by SansPapyrus683** [comment]
and this is the updated code?

**Comment by leoleo28** [comment]
> and this is the updated code?

Yes, it is updated code

**Review by SansPapyrus683 — APPROVED** [review]

**Review by ryanchou-dev — APPROVED** [review]
good!

**Pull request merged**